controllers/catalog_predicates.go

```diff
@@ -0,0 +1,115 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+	operatorv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Predicate for reconciling operators when available (i.e. ready) catalogsources on cluster change
+type catalogReadyTransitionPredicate struct {
+	predicate.Funcs
+	catalogReady map[string]bool
```

Comment:
I'm hesitant to include a cache of catalog statuses, could you explain why this is needed?

Reply:
I added this as while I was testing for generic events on a cluster, this isn't a necessary piece of the code. Thanks for pointing that out, I'll clean that up.